### DISCUSSION TITLE
Is there a way to make the makeOne and make functions of the EntityFactory not persist?

### DISCUSSION THREAD

**Discussion by squivix** [mikro-orm/mikro-orm#3932] [Q&A] [answered]
As I understand it, `makeOne` and `make` do not flush (no changes are written to db), but they do call `em.persist()`, so with the next `em.flush()` call a new entity is written to the db.

I am using an `EntityFactory` with the intent of just generating sample data, not saving to the database. While I could just call `em.clear()` after every `makeOne` or `make` call, I'm looking for a less cumbersome way. I tried overriding them to call `em.clear()` but it seems `em` is private in the `EntityFactory`

**Comment by B4nan** [comment, marked as answer]
You could just pass in a fork yourself and never flush it:

```ts
const factory = new AuthorFactory(em.fork());
// now the entities will get persisted to that fork, which will never get flushed (unless you use `make` methods).
```

> but it seems em is private

That has no limitations on runtime, its just "typescript private", not "native private", you can always just do `(em as any)`.

**Reply by B4nan** [reply]
But looking at the code, you basically want what the `makeEntity` method already does, I don't mind making it public. Worth mentioning that in current versions even `em.create()` (which is used under the hood in the seeder factory) is calling `em.persist()` automatically by default, so I also made the `makeEntity()` specifically disable that feature.

https://github.com/mikro-orm/mikro-orm/commit/bb8f1b03f7c25811f87311e997646bf793b52a13
